distribution/sender/routes/sync-response-processor.xml

```diff
@@ -0,0 +1,66 @@
+<routes xmlns="http://camel.apache.org/schema/spring">
+    <route id="sync-response-processor" errorHandlerRef="shutdownErrorHandler">
+
+        <from uri="scheduler:sync_response_processor?initialDelay={{sync-response-processor.initial.delay}}&amp;delay={{sync-response-processor.delay}}" />
+
+        <log message="Fetching next {{sync-response-processor.max.fetch.results}} sync responses" loggingLevel="DEBUG" />
+
+        <toD uri="jpa:SenderSyncResponse?query=SELECT r FROM SenderSyncResponse r WHERE r.status = 'NEW' ORDER BY r.dateSent, r.id ASC&amp;maximumResults={{sync-response-processor.max.fetch.results}}" />
+
+        <choice>
+            <when>
+                <simple>${body.size()} > 0</simple>
+                <log message="Fetched ${body.size()} sender sync responses" />
+                
+                <split parallelProcessing="false">
+                    <simple>${body}</simple>
+                    
+                    <setProperty name="syncResponse">
+                        <simple>${body}</simple>
+                    </setProperty>
+
+                    <log message="Processing sender sync response -> ${exchangeProperty.syncResponse}" />
+                    
+                    <toD uri="jpa:SenderSyncMessage?query=SELECT m FROM SenderSyncMessage m WHERE m.messageUuid='${exchangeProperty.syncResponse.messageUuid}'" />
+                    
+                    <choice>
+                        <when>
+                            <simple>${body.size()} > 0</simple>
+                            <log message="Fetched ${body.size()} Sender sync messages with uuid ${exchangeProperty.syncResponse.messageUuid}" />
+            
+                            <log loggingLevel="DEBUG" message="Sender sync message(s) found -> ${body}" />
+                            
+                            <log loggingLevel="DEBUG" message="Removing Sender sync message(s) with uuid ${exchangeProperty.syncResponse.messageUuid}" />
+            
+                            <toD uri="jpa:SenderSyncMessage?query=DELETE FROM SenderSyncMessage WHERE messageUuid = '${exchangeProperty.syncResponse.messageUuid}'" />
+            
+                            <log loggingLevel="INFO" message="Successfully removed Sender sync message(s) with uuid ${exchangeProperty.syncResponse.messageUuid}" />
+                        </when>
+                        <otherwise>
+                            <log loggingLevel="INFO" message="No Sender sync message was found with uuid ${exchangeProperty.syncResponse.messageUuid}" />
+                        </otherwise>
+                    </choice>
+                    
+                    <log message="Updating sync response status to processed" />
+
+                    <script>
+                        <spel>
+                            #{getProperty('syncResponse').markAsProcessed()}
```

Comment:
After a response is processed, delete it instead, there is no reason for keeping it around

Reply:
Implemented.